tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/object/ObjectTest.java

```diff
@@ -371,14 +371,18 @@ public void testObjectSelfreferenceWithDefaultable() {
         Assert.assertEquals(returns, 89L);
     }
 
-    @Test(description = "Test function references from an object")
-    public void testFunctionReferencesFromObjects() {
+    @Test(description = "Test function references from an object", dataProvider = "functionReferencesFromObjectTests")
+    public void testFunctionReferencesFromObjects(String functionName) {
         CompileResult compileResult = BCompileUtil.compile("test-src/object/object_function_pointer.bal");
```

Comment:
Shall we move `compileResult` inside the `setup()` method? It will compile multiple times for each call from the data provider.